bottlecap/src/metrics/aggregator.rs

```diff
@@ -274,56 +324,92 @@ impl<const CONTEXTS: usize> Aggregator<CONTEXTS> {
+        for (name, kind) in entry.tag() {
+            let resource = datadog::Resource {
+                name: name.as_str(),
+                kind: kind.as_str(),
             };
+            resources.push(resource);
+        }
+        let kind = match entry.kind {
+            metric::Type::Count => datadog::DdMetricKind::Count,
+            metric::Type::Gauge => datadog::DdMetricKind::Gauge,
+            metric::Type::Distribution => unreachable!(),
+        };
+        let point = datadog::Point {
+            value: entry.metric_value.get_value(),
+            // TODO(astuyve) allow user to specify timestamp
+            timestamp: time::SystemTime::now()
+                .duration_since(time::UNIX_EPOCH)
+                .expect("unable to poll clock, unrecoverable")
+                .as_secs(),
+        };
+
+        let mut final_tags = Vec::new();
+        // TODO
+        // These tags are interned so we don't need to clone them here but we're just doing it
+        // because it's easier than dealing with the lifetimes.
+        if let Some(tags) = entry.tags {
+            final_tags = tags
+                .split(',')
+                .map(std::string::ToString::to_string)
+                .collect();
+        }
+        final_tags.append(&mut self.tags_provider.get_tags_vec());
+        Some(DatadogMetric {
+            metric: entry.name.as_str(),
+            resources,
+            kind,
+            points: [point; 1],
+            tags: final_tags,
+        })
+    }
 
-            let mut final_tags = Vec::new();
-            // TODO
-            // These tags are interned so we don't need to clone them here but we're just doing it
-            // because it's easier than dealing with the lifetimes.
-            if let Some(tags) = entry.tags {
-                final_tags = tags
-                    .split(',')
-                    .map(std::string::ToString::to_string)
-                    .collect();
+    #[allow(clippy::cast_precision_loss)]
+    #[must_use]
+    pub fn to_series_serialized(&self) -> Vec<u8> {
+        let mut buffer: Vec<u8> = Vec::with_capacity(self.max_content_size_bytes);
+        let mut count_entries = 0;
+        for entry in &self.map {
+            if entry.kind == metric::Type::Distribution {
+                continue;
             }
-            final_tags.append(&mut self.tags_provider.get_tags_vec());
-            let metric = datadog::Metric {
-                metric: entry.name.as_str(),
-                resources,
-                kind,
-                points: [point; 1],
-                tags: final_tags,
+            let metric = self.build_metric(entry);
+            let serialized_metric = match serde_json::to_vec(&metric) {
+                Ok(serialized_metric) => serialized_metric,
+                Err(e) => {
+                    error!("failed to serialize metric: {:?}", e);
+                    continue;
+                }
```

Comment:
Can this code be changed to another pattern? Something like an `if let`? It just looks very strange

Reply:
can you give me an example?
I am aware of if let with a Some type, but with Result and using the error I don't know whether it is possible

Reply:
Maybe an `unwrap_or_else` might work here:
```
let serialized_metric = serde_json::to_vec(&metric).unwrap_or_else(|e| {
    error!("failed to serialize metric: {:?}", e);
    continue;
});
```
?